_minutes/2023-09-12-wecg-tpac.md

```diff
@@ -0,0 +1,242 @@
+ * [rew] Maybe I'm missing some of the details, but wouldn't the browser be fetching this data from the domain, not the extension?
+ * [devlin] Yes. But me, a malicious extension, can inject whatever I want on a website after the proper authentication has completed. A script can reach in and extract the data it wants from the page.
+ * [rew] Doesn't that break the WebAuthN security model?
+ * [devlin] Yes
+ * [nick] You can't intercept the signing event, but you can intercept the response coming out which is a valid attack.
+ * [carlos] Wouldn't that mean that it's more secure for an extension to perform this work because the extension would get the PR ID rather than a website? Since the PR ID would only be accessible by extensions and linked extensions. While a PR ID from a website could be received by extension content scripts.
+ * [devlin] From the POV of security (with Dashlane as an example), storing data in the extension origin would actually be an improvement, because other extensions cannot access other extensions' origins. Of course it depends on the threat model; most don't consider malicious extensions part of the threat model.
+ * [nick] So, I go to dashlane.com and a malicious extension has full access to the page. You wouldn't be able to extract a passkey from the page. Same for bitwarden, etc. You couldn't access the key
+ * [??] it would be the assertion
+ * [devlin] It wouldn't let you get the raw key, but nothing here is about the raw key.
+ * [tomislav] What does this actually mean? What's the actual threat model? It already has access to the attestation. If you can access the entire vault data on the page, then a malicious extension has full access to this data. So what are we actually trying to protect?
+ * [nick] If you run malicious code on your own client, we have no meaningful way of interdicting that.
+ * [devlin] That's our stance as well. If you run a malicious extension on a site it has access to, well, it has access. To me, this browser validation step is meaningless.
+ * [tomislav] It doesn't buy us any security benefit.
+ * [devlin] right, this is just an alternative to providing a host permission. Extensions can get exactly the same functionality by just requesting a host permission.
+ * [tim c] On the web platform we need both sides. On the extension, we only need the name binding (with the assumption that the security isn't broken)
+ * [devlin] I can understand the name binding use case. In that case we take out the browser validation, and bikeshedding aside, allow extensions to specify a RP ID in the manifest.json file, provided that the extension is granted host permissions.
+ * [nick] So, the attacker gains a response, they can use it once. Are we assuming continued access?
+ * [tim c] Not with a passkey. It's always 0.
+ * &lt;much webauthn terminology>
+ * [oliver] I don't think it's that hard. As the extension, if you create the private key, you restrict access to it.
```

Comment:
```suggestion
 * [oliver] I don't think it's that hard to get full access to the key. If a malicious extension is present at signup it can use its own key in the `navigation.credentials.create` response.
```